src/api/firebase.ts

```diff
@@ -49,35 +56,54 @@ export const fetchStats = async (chainId: number): Promise<Stats> => {
   }
 };
 
-export const fetchPrograms = async (chainId: number) => {
+export const fetchPrograms = async (
+  chainId: number
+): Promise<ExtendedRewardProgram[]> => {
   try {
     const statsRef = db.collection('rewardPrograms').doc(chainId.toString());
     const doc = await statsRef.get();
+    if (!doc.data()) {
+      throw new Error('document not found');
+    }
     const response = doc.data();
     return Object.values(response as {});
-  } catch (err) {
-    console.log(err);
+  } catch (err: any) {
+    const errMessage =
+      err && err.message ? err.message : 'Failed to get reward programs';
+    throw new Error(errMessage);
   }
 };
 
-export const fetchRewards = async (chainId: number) => {
+export const fetchRewards = async (
+  chainId: number
+): Promise<ProgramRewards> => {
   try {
     const statsRef = db.collection('programRewardsUSD').doc(chainId.toString());
     const doc = await statsRef.get();
-    const response = doc.data();
+    if (!doc.data()) {
+      throw new Error('document not found');
+    }
+    const response = doc.data() as ProgramRewards;
     return response;
-  } catch (err) {
-    console.log(err);
+  } catch (err: any) {
+    const errMessage =
+      err && err.message ? err.message : 'Failed to get rewards usd';
+    throw new Error(errMessage);
   }
 };
 
-export const fetchStaked = async (chainId: number) => {
+export const fetchStaked = async (chainId: number): Promise<ProgramStaked> => {
   try {
     const statsRef = db.collection('programStakedUSD').doc(chainId.toString());
     const doc = await statsRef.get();
-    const response = doc.data();
+    if (!doc.data()) {
+      throw new Error('document not found');
+    }
+    const response = doc.data() as ProgramStaked;
     return response;
-  } catch (err) {
-    console.log(err);
+  } catch (err: any) {
+    const errMessage =
+      err && err.message ? err.message : 'Failed to get staked programs';
```

Comment:
Minor change to match the rewards error message? 😄 
```suggestion
      err && err.message ? err.message : 'Failed to get staked usd';
```